psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -2,130 +2,135 @@
-            return;
-        }
-        String clientType = args[0];
-        String constraint = args[1];
-
-        //connect to host server
-        DataInputStream fromHost;
-        DataOutputStream toHost;
-        while (true) {
-            try {
-                System.out.println("Attempting to connect to localhost:9999");
-                Socket socket = new Socket();
-                socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 9999), 5000);
-                fromHost = new DataInputStream(socket.getInputStream());
-                toHost = new DataOutputStream(socket.getOutputStream());
-                break;
-            } catch (Exception e) {
-                System.err.println("Retrying: " + e.getMessage());
-                Thread.sleep(2000);
+    @CommandLine.Option(names = {"-u", "--url"},
+                required = true,
+                description = "URL of the web host running the enclave.")
+        private String url;
+
+        @CommandLine.Option(names = {"-c", "--constraint"},
+                required = true,
+                description = "Enclave constraint which determines the enclave's identity and whether it's acceptable to use.",
+                converter = EnclaveConstraintConverter.class)
+        private EnclaveConstraint constraint;
+
+    /**
+     * Use Kryo to serialize inputs from client to enclave
+     */
+    private Output serializeMessage(InputData listOfCreditCardNumbers){
+        Kryo kryo = new Kryo();
+        Output output = new Output(new ByteArrayOutputStream());
+        kryo.register(InputData.class, new InputDataSerializer());
+        kryo.writeObject(output, listOfCreditCardNumbers);
+        output.close();
+        return output;
+    }
+
+    @Override
+    public Void call() throws Exception {
+        //a new private key is generated. Enclave Client is created using this private key and constraint.
+        //a corresponding public key will be used by the enclave to encrypt data to be sent to this client
+        EnclaveClient enclaveClient = new EnclaveClient(constraint);
+
+        try (WebEnclaveTransport transport = new WebEnclaveTransport(url);
+             EnclaveClient client = enclaveClient)
+        {
+            //retrieve the enclaveInstanceInfo object, i.e. the remote attestation object and verify it against the
+            //constraint
+            client.start(transport);
+
+            //collect merchants and service providers credit card numbers list
+            InputData inputData = getInputData();
+
+            byte[] requestMailBody = serializeMessage(inputData).getBuffer();
+
+            //client will send its credit card list to enclave and wait for other client to send their list
+            EnclaveMail responseMail = client.sendMail(requestMailBody);
+
+            //responseMail is null till enclave doesn't reply back to the client
+            if(responseMail == null) {
+                do
+                {
```

Comment:
Minor: Move the `{` up to the `do` line, to be consistent with the rest of the code.

Reply:
done